docs/Developers/NextGen_Core.rst

```diff
@@ -0,0 +1,130 @@
+NextGen-Core
+============
+
+This document provides a brief overview of coala's NextGen-Core.
+coala's NextGen-Core comes with the promise of eliminating the dependencies
+amongst bears (by removing the queuing machanism) and a better performance
+(by utilising asynchronous I/O and process pool).
+
+What has changed?
+-----------------
+
+``Global Bears`` are now called ``Project Bears`` and ``Local Bears`` are
+now known as ``File Bears``. Both the ``ProjectBear`` and the ``FileBear``
+classes inherit from the base class ``Bear.py``.
```

Comment:
maybe it should be mentioned - from the base class `coalib.bears.Bear.py` or `coalib.core.Bear.py`?

Reply:
I'm not sure if that's necessary. IIRC there is only one `Bear.py` so there won't any need to mention the whole path of the file and if a developer wants to know where it resides they can always take a look at the api docs

Reply:
there are two :confused:

Reply:
You are right. I just did a git grep and saw 2 of them. I'll change it ASAP.

Reply:
@manankalra done and repushed